fuel-tx/src/receipt/receipt_std.rs

```diff
@@ -330,6 +330,56 @@ impl io::Read for Receipt {
                 bytes::store_number_at(buf, S::layout(S::LAYOUT.len), *len);
                 bytes::store_at(buf, S::layout(S::LAYOUT.digest), digest);
             }
+            Receipt::Mint {
+                sub_id,
+                contract_id,
+                val,
+                pc,
+                is,
+            } => {
+                type S = MintSizes;
+                let buf: &mut [_; S::LEN] = buf
+                    .get_mut(..S::LEN)
+                    .and_then(|slice| slice.try_into().ok())
+                    .ok_or(bytes::eof())?;
+
+                bytes::store_number_at(
+                    buf,
+                    S::layout(S::LAYOUT.repr),
+                    ReceiptRepr::Mint as u8,
+                );
+
+                bytes::store_at(buf, S::layout(S::LAYOUT.sub_id), sub_id);
```

Comment:
Why do we put sub id before contract id?

Reply:
It is layout from the specification=)
<img width="651" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/fbc2dce7-49bd-4b3f-abe2-171d57226529">